wrappers/wrapper.rb

```diff
@@ -572,6 +572,85 @@ def detect_unfeasible_services(vrp)
       unfeasible
     end
 
+    def sequence_relation_with_no_reachable_vehicle(unfeasible, vrp, relation)
+      return false unless Models::Relation::POSITION_TYPES.include?(relation.type)
+
+      vrp.vehicles.each{ |v|
+        unreachable_service_id = last_service_reachable_sequence(vrp, v, relation.linked_services)
+        next unless unreachable_service_id
+
+        service_met = false
+        relation.linked_services.each{ |service|
+          service_met = true if service.id == unreachable_service_id
+          next unless service_met
+
+          service.vehicle_compatibility[v.id] = false
+        }
+      }
+
+      previous_removed = false
+      to_delete_services = []
+      relation.linked_services.each{ |service|
+        previous_removed ||= vrp.vehicles.none?{ |vehicle|
+          service.vehicle_compatibility[vehicle.id]
+        }
+
+        if previous_removed
+          if Models::Relation::ALL_OR_NONE_RELATIONS.include?(relation.type)
+            to_delete_services = relation.linked_services
+            break false
+          else
+            to_delete_services << service
+          end
+        end
+      }
+
+      relation.linked_services -= to_delete_services
+
+      to_delete_services.each{ |service|
+        add_unassigned(unfeasible, vrp, service, "Service belongs to a relation of type #{relation.type} which makes it infeasible")
+      }
+    end
+
+    def last_service_reachable_sequence(vrp, vehicle, services)
+      vehicle_timewindow = vehicle.timewindow || Models::Timewindow.new(start: 0)
+      vehicle_duration = [vehicle.duration, vehicle_timewindow.safe_end(vehicle.cost_late_multiplier&.positive?) - vehicle_timewindow.start].compact.min
+      successive_activities = services.map{ |service|
+        return service.id if service.vehicle_compatibility[vehicle.id] == false
+        [service.id, service.activity && [service.activity] || service.activities]
+      }
+      return nil if successive_activities.all?{ |_id, acts| acts.any?{ |a| a.timewindows.none?(&:end) } }
```

Comment:
What about the case an order relation is infeasible due to vehicle capacity? We normally handle capacity violating services with detect_unfeasible. Don't we want to generalise all this logic to these ALL_OR_NOTHING or POSITION_TYPE relations. 

This may be out-of-scope of this PR and maybe better placed in the generalisation ticket -- can you share me the link or add this message to the ticket (unless you already have this in the ticket, grands-esprit) so that we keep it in mind during generalization.